### ISSUE TITLE
Discrepancy in output shape for batch_norm inference mode between CUDA and CPU

### ISSUE THREAD

**Issue by IvanYashchuk** [pytorch/pytorch#85960]
### 🐛 Describe the bug

For CPU inputs `torch.native_batch_norm` returns 0-sized outputs, but for CUDA inputs it returns a copy of `running_mean` argument and inverse of standard deviation.
What's the reason for this discrepancy and what's the correct output? Can we align the behavior of two backends?

CPU code path:
https://github.com/pytorch/pytorch/blob/8f4edf1e1dc9419f0bab66a67c8f149d7b53fc25/aten/src/ATen/native/Normalization.cpp#L684-L687

CUDA code path:
https://github.com/pytorch/pytorch/blob/fe87ae692f813934d1a74d000fd1e3b546c27ae2/aten/src/ATen/native/cuda/Normalization.cu#L440-L443

### Versions

Latest master.

**Comment by jjsjann123** [comment]
cc'ing @ptrblck

**Comment by albanD** [comment]
Agreed with Natalia on this!

**Comment by jjsjann123** [comment]
It's unfortunate that we don't have consistency check for BN api behavior. Patch on cuda kernel behavior should be easy. We can try to put in more robust test on the APIs. I'm thinking opinfo test, which is supposed to regulate these.

cc'ing @xwang233

**Comment by lezcano** [comment]
@srossross did your checks catch this one?

**Comment by srossross** [comment]
I've rebased from master and now catches this discrepancy
